src/gsSolver/gsAdditiveOp.h

```diff
@@ -0,0 +1,127 @@
+/** @file gsAdditiveOp.h
+
+    @brief Allows to set up additive Schwarz type preconditioners
+
+    This file is part of the G+Smo library.
+
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s): S. Takacs
+*/
+
+#pragma once
+
+#include <gsSolver/gsPreconditioner.h>
+#include <gsSolver/gsMatrixOp.h>
+#include <gsCore/gsMultiPatch.h>
+
+namespace gismo
+{
+
+/// @brief Generic preconditioner which applies an arbitrary linear operator to the residual.
+///
+/// \code{.cpp}
+///    gsLinearOperator<>::Ptr pc = gsAdditiveOp<>::make( transfers, ops );
+/// \endcode
+///
+/// is equivalent to
+///
+/// \code{.cpp}
+///    gsSumOp<>::Ptr pc = gsSumOp<>::make();
+///
+///    for (index_t i=0; i<transfers.size(); ++i)
+///       s->addOperator(
+///          gsProductOp<>::make(
+///             makeMatrixOp(transfers[i].transpose()),
+///             ops[i],
+///             makeMatrixOp(transfers[i])
+///          )
+///       );
+/// \endcode
+///
+/// but much faster.
+///
+/// @ingroup Solvers
+
+template<class T>
+class gsAdditiveOp : public gsLinearOperator<T>
+{
+    typedef typename gsLinearOperator<T>::Ptr   OpPtr;
+    typedef std::vector<OpPtr>   OpContainer;
+    typedef gsSparseMatrix<T,RowMajor>   Transfer;
+    typedef std::vector<Transfer>   TransferContainer;
+
+public:
+
+    /// Shared pointer
+    typedef memory::shared_ptr<gsAdditiveOp> Ptr;
+
+    /// Unique pointer
+    typedef memory::unique_ptr<gsAdditiveOp> uPtr;
+
+    /// Default Constructor
+    gsAdditiveOp() : m_transfers(), m_ops() {}
+
+    /// Constructor
```

Comment:
Please document each argument (same for make(..) and os on)
eg. what is the size of the m_transfers that is expected ?